### PR TITLE
Add config file API

### PR DESCRIPTION
**Pull request by axic**
Implements #374. Depends on #376.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -418,10 +421,18 @@ var run = function () {
     }
   })
 
+  // convert old browser-solidity
```
Comment (axic) [inline src/app.js]: Do we really need to carry over this single config option? Probably not such a big deal.

Comment (axic) [inline src/app.js]: #377 removes a clean way to carry this on, I would say it is not a big deal losing this information

Comment (chriseth) [inline src/app.js]: Carrying over config is not too important.

File: src/app/config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+var CONFIG_FILE = utils.fileKey('.browser-solidity.json')
+
+function Config (storage) {
+  this.keys = {}
+
+  // load on instantiation
+  var config = storage.get(CONFIG_FILE)
+  if (config) {
+    this.keys = JSON.parse(config)
```
Comment (chriseth) [inline src/app/config.js]: catch exception?

File: src/app/config.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+var utils = require('./utils')
+
+var CONFIG_FILE = utils.fileKey('.browser-solidity.json')
+
+function Config (storage) {
+  this.keys = {}
```
Comment (chriseth) [inline src/app/config.js]: This does not only contains the keys but also the values.

File: src/app.js (hunk 1 above)
Comment (axic) [inline src/app.js]: Realized that I still want to keep there the removal of the key, so for the moment this can stay. In #377 it will only delete key.

**Comment by axic** [comment]
@chriseth updated

File: src/app/config.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+function Config (storage) {
+  this.items = {}
+
+  // load on instantiation
+  try {
+    var config = storage.get(CONFIG_FILE)
+    if (config) {
+      this.items = JSON.parse(config)
+    }
+  } catch (exception) {
```
Comment (chriseth) [inline src/app/config.js]: We should really set the editor into some fail-safe mode if this happens. Currently, we do not yet store much valuable information in the config, but as soon as we do, we should not overwrite existing config files which are malformed without the user's confirmation.

Comment (axic) [inline src/app/config.js]: Can you add an issue for that so we don't forget?

**Pull request merged**